src/crypto.ts

```diff
@@ -1,17 +1,27 @@
-import { Handshake } from "./handshake";
 import { Buffer } from "buffer";
+import { Handshake } from "./handshake";
 
 interface ReturnEncryptionWrapper {
   (source: Iterable<Uint8Array>): AsyncIterableIterator<Uint8Array>;
 }
 
+const maxPlaintextLength = 65519;
+
 // Returns generator that encrypts payload from the user
 export function encryptStream(handshake: Handshake): ReturnEncryptionWrapper {
   return async function * (source) {
     for await (const chunk of source) {
       const chunkBuffer = Buffer.from(chunk);
```

Comment:
I'm pretty sure this is creating large overhead because it's allocating new memory and copy chunk data.
I think following code reuses existing memory:
```javascript
Buffer.from(chunk.buffer, chunk.byteOffset, chunk.length)
```

Ref: https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_arraybuffer_byteoffset_length